include/glow/Support/TensorPool.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_TENSORPOOL_H
+#define GLOW_TENSORPOOL_H
+
+#include "glow/Base/Tensor.h"
+
+#include <atomic>
+#include <iostream>
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+namespace glow {
+
+class TensorPool final {
+private:
+  struct TypeHash {
+    size_t operator()(const Type &t) const { return t.equals_hash(); }
+  };
+
+  struct TypeEquals {
+    bool operator()(const Type &a, const Type &b) const { return a.isEqual(b); }
+  };
+  /// A stack of available Tensors per Type.
+  std::unordered_map<Type, std::vector<Tensor *>, TypeHash, TypeEquals> pools_;
```

Comment:
I noticed that you did not use the random-access property of the `std::vector`. Could it be a `std::queue`? Or `std::stack`?